src/ripple/shamap/SHAMapMissingNode.h

```diff
@@ -66,6 +69,13 @@ class SHAMapMissingNode : public std::runtime_error
               "Missing Node: " + to_string(t) + ": id " + to_string(id))
     {
     }
+
+    SHAMapMissingNode(uint256 const& id)
+        : std::runtime_error(
+              "Missing Node: " + to_string(SHAMapType::UNKNOWN) + ": hash " +
```

Comment:
This new enumeration value seems pointless. We can just use the constant `to_string` return value directly here. If we're going to add an enumeration value, then shouldn't we use the constructor immediately before this one instead? That said, I'd prefer to not use this class at all and just directly throw a `std::runtime_error`.

Reply:
If you want to go through all the code paths where this would bubble up to make sure it's caught, then pick the very most best exception, then go for it. I'm assuming that SHAMapMissingNode is probably the one that already gets caught in some sensible way to not dump core, but am not 100% certain.

Reply:
The way I implemented this makes sense to me from a consistency perspective, that each instance has a type, which is then converted to string -- even though the caller isn't specifying the type.

Reply:
The caller is specifying the type in the existing constructors. Why not use one of them?